### ISSUE TITLE
False positive on import when it is shadowed by another module variant being imported

### ISSUE THREAD

**Issue by kraklin** [jfmengels/elm-review-unused#29]
When you have the same name of type and name of variant
AND you expose the variants with (..) and the type explicitly
THEN elm-review wants the first one to be removed even though it is used and without it the program won't compile.

**SSCCE (Short Self-Contained Correct Example)**
https://github.com/kraklin/elm-review-unused-15-bug-sscce

**Expected behavior**
I'd expect this is not false positive as the removal of the code marked as unused results in app not being buildable.

**Comment by sporto** [comment]
This just happend to me:

I have

```
import Api.Enum.Interval exposing (Interval(..))
import Time exposing (Month, Posix)

type alias MonthInYear =
    { year : Int, month : Month }
```

The `Interval` happens to expose `Month` as a variant of `Interval`

Elm review says:

```
NoUnused.Variables: Imported type `Month` is not used
```

**Comment by jfmengels** [comment]
I've opened a PR: #34

I'd love for someone to try it out!
```bash
elm-review --template jfmengels/elm-review-unused/preview#issue-29
```

**Comment by kraklin** [comment]
I have tested it out both on the SSCCE and our code before the workaround on this bug and I can't reproduce the problem anymore, so I'd say the problem is gone. Nice work! 👏❤️

**Comment by jfmengels** [comment]
Released in 1.1.6!
